src/exceptions.rs

```diff
@@ -359,7 +359,9 @@ Python::with_gil(|py| {
 impl_native_exception!(
     PyBaseException,
     PyExc_BaseException,
-    native_doc!("BaseException")
+    native_doc!("BaseException"),
+    ffi::PyBaseExceptionObject,
+    #checkfunction=ffi::PyExceptionInstance_Check
```

Comment:
This is just an optimization to use a "fast" check for `isinstance` for `BaseException`. I'll split this out into a separate PR too.